opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchScrollRequest.java

```diff
@@ -87,8 +90,12 @@ public OpenSearchResponse search(Function<SearchRequest, SearchResponse> searchA
       openSearchResponse = searchAction.apply(searchRequest());
     }
     setScrollId(openSearchResponse.getScrollId());
+    FetchSourceContext fetchSource = this.sourceBuilder.fetchSource();
+    List<String> includes = fetchSource != null && fetchSource.includes() != null
+        ? Arrays.asList(this.sourceBuilder.fetchSource().includes())
+        : List.of();
```

Comment:
Probably it should be done before search (move 6 lines up)

Reply:
Why? It shouldn't be affected by the search.